dashboard/legacy/middleware/files_api.rb

```diff
@@ -645,7 +645,11 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
 
     filename.downcase! if endpoint == 'files'
     begin
-      get_bucket_impl(endpoint).new.list_versions(encrypted_channel_id, filename, with_comments: request.GET['with_comments']).to_json
+      versions = get_bucket_impl(endpoint).new.list_versions(encrypted_channel_id, filename, with_comments: request.GET['with_comments'])
+      return versions.to_json if owns_channel?(encrypted_channel_id)
+      # only return the latest version if the user does not own the channel
+      latest_version = versions.select {|version| version[:isLatest]}
+      return latest_version.to_json
```

Comment:
tiny nit: I didn't think this was an array at first because of the name `version`, so I was briefly concerned that this returned an array in one scenario and an object in another. Not sure if removing the variable name and making 651 & 552 one line makes it more or less clear.

Reply:
I think you're right, I'll update it